// @mplp/sdk-ts/src/core/validators/index.ts
/**
 * © 2025 Bangshi Beijing Network Technology Limited Company
 * Licensed under the Apache License, Version 2.0.
 *
 * This file is part of the MPLP reference implementation.
 * It is NOT part of the frozen protocol specification.
 */
/**
 * © 2025 Bangshi Beijing Network Technology Limited Company
 * Licensed under the Apache License, Version 2.0.
 *
 * This file is part of the MPLP reference implementation.
 * It is NOT part of the frozen protocol specification.
 */
/**
 * © 2025 Bangshi Beijing Network Technology Limited Company
 * Licensed under the Apache License, Version 2.0.
 *
 * This file is part of the MPLP reference implementation.
 * It is NOT part of the frozen protocol specification.
 */
/**
 * © 2025 Bangshi Beijing Network Technology Limited Company
 * Licensed under the Apache License, Version 2.0.
 *
 * This file is part of the MPLP reference implementation.
 * It is NOT part of the frozen protocol specification.
 */
export interface ValidationResult {
    ok: boolean;
    errors: any[];
}

export function validateContext(data: any): ValidationResult {
    return { ok: true, errors: [] };
}

export function validatePlan(data: any): ValidationResult {
    return { ok: true, errors: [] };
}

export function validateConfirm(data: any): ValidationResult {
    return { ok: true, errors: [] };
}

export function validateTrace(data: any): ValidationResult {
    return { ok: true, errors: [] };
}
